src/genericitems/list.cpp

```diff
@@ -2,293 +2,186 @@
 // Copyright (c) 2021 Team Dissolve and contributors
 
 #include "genericitems/list.h"
+#include <cassert>
+#include <classes/partialset.h>
+#include <fmt/core.h>
+#include <typeindex>
+
+GenericList::GenericList()
+{
+    registerBroadcasters();
+    registerDeserialisers();
+    registerProducers();
+    registerSerialisers();
+}
 
 // Clear all items (except those that are marked protected)
 void GenericList::clear()
 {
-    GenericItem *item = items_.first(), *nextItem;
-    while (item)
-    {
-        nextItem = item->next();
-        if (!item->hasFlag(GenericItem::ProtectedFlag))
-            items_.remove(item);
-        item = nextItem;
-    }
+    for (auto it = items_.begin(); it != items_.end();)
+        if (!(std::get<ItemData::Flags>(it->second) & GenericList::ProtectedFlag))
+            items_.erase(it);
+        else
+            ++it;
```

Comment:
```suggestion
    items_.erase(std::remove_if(items_.begin(), items_.end(), [=](auto& [key, value]){ return !(std::get<ItemData::Flags>(value) & GenericList::ProtectedFlag);}, items_.end());
```

Might need to take a closer look at what I've included in with the lambda, but I think I got that right.

Reply:
For some reason I was under the impression that `std::map` didn't have an `erase()` overload that took begin and end iterators.  It's been a tough week!